workers/unity/Packages/com.improbable.gdk.playerlifecycle/Systems/PlayerCreation/SendCreatePlayerRequestSystem.cs

```diff
@@ -29,18 +35,29 @@ protected override void OnCreateManager()
             commandSystem = World.GetExistingManager<CommandSystem>();
             logDispatcher = workerSystem.LogDispatcher;
 
+            playerCreatorEntityIds = new List<EntityId>();
+
             initializationGroup = GetComponentGroup(
                 ComponentType.ReadOnly<WorkerEntityTag>(),
                 ComponentType.ReadOnly<OnConnected>()
             );
         }
 
-        public void RequestPlayerCreation(byte[] serializedArguments = null)
+        public bool RequestPlayerCreation(byte[] serializedArguments = null)
         {
             serializedArgumentsCache = serializedArguments;
-            var request = new CreatePlayerRequestType(serializedArguments);
-            var createPlayerRequest = new PlayerCreator.CreatePlayer.Request(playerCreatorEntityId, request);
-            commandSystem.SendCommand(createPlayerRequest);
+
+            var playerCreatorCount = playerCreatorEntityIds.Count;
+            if (playerCreatorCount > 0)
+            {
+                commandSystem.SendCommand(new PlayerCreator.CreatePlayer.Request(
+                    playerCreatorEntityIds[Random.Range(0, playerCreatorCount)],
+                    new CreatePlayerRequestType(serializedArgumentsCache)));
+                return true;
+            }
+
+            Debug.LogWarning("Unable to send player creation request: no player creator entities found yet.");
```

Comment:
I don't think this warning is particularly useful here - as a user I'd rather you queued my request and send it when it was found.

This warning should be thrown (as an error) if the query fails to find any matching entities (more than once).